grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -166,8 +167,31 @@ class ProductAvailabilityService {
             }
             picklistItems {
                 if (product) {
+                    or {
+                        requisitionItem {
+                            eq("product", product)
+                        }
+                        orderItem {
+                            eq("product", product)
+                        }
+                    }
+                }
+                or {
+                    orderItem {
+                        isNotNull("id")
+                    }
                     requisitionItem {
```

Comment:
I'm not sure if criteria works like that, I think the `requisitionItem` should be inside the `and`

Reply:
I'll try to rewrite it as a query.